crates/sparrow-instructions/src/evaluators/macros.rs

```diff
@@ -297,7 +297,91 @@ macro_rules! create_typed_evaluator {
     }};
 }
 
+/// Create a `Box<dyn Evaluator>` for a map instruction.
+///
+/// The `$evaluator` must be a `struct` that takes a generic datatype T
+///  and which implements `Evaluator`.
+///
+/// The `$info` should be a `StaticInfo`, which contains information relevant
+/// to creating a new evaluator.
+/// # Example
+///
+/// ```no_run
+/// struct GetStringToPrimitiveEvaluator<T: ArrowPrimitiveType>;
+/// impl<T: ArrowPrimitiveType> Evaluator for GetStringToPrimitiveEvaluator<T> { ... }
+///
+/// fn create_evaluator(args: Vec<StaticArg>, result_type: &DataType) -> anyhow::Result<Self> {
+///   create_map_evaluator!(key_type, map_value_type, GetStringToPrimitiveEvaluator, info)
+/// }
+/// ```
+macro_rules! create_map_evaluator {
+    ($key_type:expr, $input_type:expr, $string_evaluator:ident, $large_string_evaluator:ident, $info:expr) => {{
+        use arrow::datatypes::*;
+        use create_primitive_evaluator;
+        match $key_type {
+            DataType::Utf8 => create_primitive_evaluator!($input_type, $string_evaluator, $info),
```

Comment:
I'd expect all the other evaluators to be changed to include `LargeUtf8` if they support `Utf8`. Eg., we should be able to do string len, substring, etc. on it, right?

Reply:
Agreed. Tracking in https://github.com/kaskada-ai/kaskada/issues/494